File: app/src/main/java/com/example/win/easy/dagger/module/BackendRequestModule.java
package com.example.win.easy.dagger.module;

import com.example.win.easy.web.request.BackendRequestService;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

@Module
public class BackendRequestModule {

    @Provides @Singleton static BackendRequestService provideBackendResourceWebService(@Named("serverUrl") String serverUrl){
        Retrofit retrofit=new Retrofit.Builder()
                .baseUrl(serverUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        return retrofit.create(BackendRequestService.class);
    }

    @Provides @Named("serverUrl") static String provideServerUrl(){
        return "http://guohere.com:9000/";
    }
}

File: app/src/main/java/com/example/win/easy/dagger/module/BackendRequestModule.kt
package com.example.win.easy.dagger.module

import com.example.win.easy.network.BackendRequestService
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import javax.inject.Named
import javax.inject.Singleton

@Module
object BackendRequestModule {

    @JvmStatic
    @Provides
    @Singleton
    fun provideBackendResourceWebService(@Named("serverUrl") serverUrl: String): BackendRequestService {
        val retrofit = Retrofit.Builder()
                .baseUrl(serverUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        return retrofit.create(BackendRequestService::class.java)
    }

    @JvmStatic
    @Provides
    @Named("serverUrl")
    fun provideServerUrl() = "http://10.0.2.2:9000/"
}